pkg/repositoryserver/controller.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2022 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package repositoryserver
+
+import (
+	"context"
+	"fmt"
+
+	osversioned "github.com/openshift/client-go/apps/clientset/versioned"
+	"github.com/pkg/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+
+	crv1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	"github.com/kanisterio/kanister/pkg/client/clientset/versioned"
+	"github.com/kanisterio/kanister/pkg/customresource"
+	"github.com/kanisterio/kanister/pkg/eventer"
+	"github.com/kanisterio/kanister/pkg/field"
+	"github.com/kanisterio/kanister/pkg/log"
+)
+
+// Controller represents a controller object for RepositoryServer resource
+type Controller struct {
+	config    *rest.Config
+	crClient  versioned.Interface
+	clientset kubernetes.Interface
+	dynClient dynamic.Interface
+	osClient  osversioned.Interface
+	recorder  record.EventRecorder
+}
+
+// NewController create controller for watching RepositoryServer resource created
+func NewController(c *rest.Config) *Controller {
+	return &Controller{
+		config: c,
+	}
+}
+
+// StartWatch watches for instances of RepositoryServer and acts on them.
+func (c *Controller) StartWatch(ctx context.Context, namespace string) error {
+	crClient, err := versioned.NewForConfig(c.config)
+	if err != nil {
+		return errors.Wrap(err, "failed to get a CustomResource client")
+	}
+	if err := checkRepositoryServersAccess(ctx, crClient, namespace); err != nil {
+		return err
+	}
+	clientset, err := kubernetes.NewForConfig(c.config)
+	if err != nil {
+		return errors.Wrap(err, "failed to get a k8s client")
+	}
+	dynClient, err := dynamic.NewForConfig(c.config)
+	if err != nil {
+		return errors.Wrap(err, "failed to get a k8s dynamic client")
+	}
```

Comment:
Can we use [kube.NewClient()](https://github.com/kanisterio/kanister/blob/master/pkg/kube/client.go#L68) and [kube.NewDynamicClient()](https://github.com/kanisterio/kanister/blob/master/pkg/kube/client.go#L68) from `github.com/kanisterio/kanister/pkg/kube` package?